File: CareConnect/CareConnect.Models/Requests/ReviewInsertRequest.cs
using System;
using System.Text.Json.Serialization;

namespace CareConnect.Models.Requests
{
    public class ReviewInsertRequest
    {
        public int UserId { get; set; }

        public string Title { get; set; } = null!;

        public string Content { get; set; } = null!;

        public int EmployeeId { get; set; }

        public int? Stars { get; set; }

        [JsonIgnore]
        public DateTime PublishDate { get; set; } = DateTime.Now;

        [JsonIgnore]
        public DateTime ModifiedDate { get; set; } = DateTime.Now;
    }
}

File: UI/careconnect_mobile/lib/models/requests/review_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'review_insert_request.g.dart';

@JsonSerializable()
class ReviewInsertRequest {
  int userId;
  String title;
  String content;
  int employeeId;
  int? stars;

  ReviewInsertRequest({
    required this.userId,
    required this.title,
    required this.content,
    required this.employeeId,
    this.stars,
  });

  factory ReviewInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$ReviewInsertRequestFromJson(json);

  Map<String, dynamic> toJson() => _$ReviewInsertRequestToJson(this);
}
